qemu/tests/virtio_fs_share_data.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import os
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context
+from virttest.remote import scp_to_remote
+
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test virtio-fs by sharing the data between host and guest.
+    Steps:
+        1. Create shared directories on the host.
+        2. Run virtiofsd daemons on the host.
+        3. Boot a guest on the host with virtiofs options.
+        4. Log into guest then mount the virtiofs targets.
+        5. Generate files or run stress on the mount points inside guest.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def create_dest_dir():
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, logging.info)
+        session.cmd(cmd_create_dest_dir % fs_dest)
+
+    def mount_target():
+        error_context.context("Mount virtiofs target %s to %s inside guest."
+                              % (fs_tag, fs_dest), logging.info)
+        session.cmd(cmd_mount_target % (fs_tag, fs_dest))
+
+    def run_dd_test():
+        error_context.context("Creating files under %s." % fs_dest, logging.info)
+        session.cmd(cmd_dd % guest_data, io_timeout)
+
+    def compare_md5():
+        error_context.context("Compare the md5 between guest and host.", logging.info)
+        md5_guest = session.cmd(cmd_md5 % guest_data, cmd_md5_timeout).strip().split()[0]
+        logging.info(md5_guest)
+        md5_host = process.run(cmd_md5 % host_data, cmd_md5_timeout).stdout_text.strip().split()[0]
+        if md5_guest != md5_host:
+            test.fail('The md5 value of host is not same to guest.')
+
+    def run_fio_test():
+        error_context.context("Run fio on %s." % fs_dest, logging.info)
+        fio = generate_instance(params, vm, 'fio')
+        try:
+            fio.run(fio_options % guest_data, io_timeout)
+        finally:
+            fio.clean()
+
+    def run_pjdfstest_test():
+        error_context.context("Run pjdfstest on %s." % fs_dest, logging.info)
+        host_path = os.path.join(data_dir.get_deps_dir('pjdfstest'), pjdfstest_pkg)
+        scp_to_remote(host_addr, port, username, password, host_path, fs_dest)
+        session.cmd(params['cmd_unpack'] % fs_dest, 180)
+        session.cmd(params['cmd_yum_deps'], 180)
+        session.cmd(params['cmd_make'] % fs_dest, io_timeout)
+        session.cmd(cmd_pjdfstest % fs_dest, io_timeout)
+
+    cmd_dd = params.get('cmd_dd')
+    cmd_md5 = params.get('cmd_md5')
+    cmd_md5_timeout = params.get_numeric('cmd_md5_timeout')
+
```

Comment:
@YongxueHong it is always good to have to default value provision, we might want to use it future, i agree the current design of test provides values through the cfg, consider the when the cfg is growing and you want to use the flexibility of certain default vaules for few params it will come handy....

Reply:
> @YongxueHong it is always good to have to default value provision, we might want to use it future, i agree the current design of test provides values through the cfg, consider the when the cfg is growing and you want to use the flexibility of certain default vaules for few params it will come handy....

Hi, @santwana  yes, i agree with you, maybe for some feature cases that need to use the flexibility of certain default vaules, but there is no such this requirement in this case.
however, actually `var = params.get('key')` is equal to `params['key']`, the key is assgined in the cfg, I could update it.